helper/identity/templating.go

```diff
@@ -86,12 +93,25 @@ func performTemplating(ns *namespace.Namespace, input string, entity *Entity, gr
 				return "", ErrTemplateValueNotFound
 			}
 			return alias.Name, nil
-		case strings.HasPrefix(trimmed, "metadata."):
-			val, ok := alias.Metadata[strings.TrimPrefix(trimmed, "metadata.")]
-			if !ok {
-				return "", ErrTemplateValueNotFound
+		case strings.HasPrefix(trimmed, "metadata.") || trimmed == "metadata":
```

Comment:
I think you said something about different modes; we should probably ensure that you can't hit the `len(1)` case below when templating in policies.

Reply:
Yep. Switched to a mode approach, and pulled apart the `metadata.` and `metadata` cases so the later should be impossible with ACL templating.